src/js/TextCopyHelper.js

```diff
@@ -1,60 +1,99 @@
-/*global require, module*/
-"use strict";
+/**global require, module*/
+'use strict';
 
 var DomDelegate = require('ftdomdelegate');
-var ToolTip = require('./ToolTip');
-var shareDom = require('./shareDom');
+var Tooltip = require('./Tooltip');
 
+/**
+ * Gets the width of a text by using a <canvas> element
+ *
+ * @param {string} text - The text to measure
+ * @param {HTMLElement} refEl - The reference element where the text will be to get the font properties
+ *
+ * @returns {number}
+ */
+function getPixelWidthOfText(text, refEl) {
+	var c = document.createElement("canvas");
+
+	if (c.getContext && window.getComputedStyle) {
+		var ctx = c.getContext("2d");
+		var refElStyle = window.getComputedStyle(refEl);
+		ctx.font = refElStyle.getPropertyValue('font-style') + " " + refElStyle.getPropertyValue('font-size') + " " + refElStyle.getPropertyValue('font-family');
+		ctx.fillText(text, 10, 100);
+		return ctx.measureText(text).width;
+	}
+
+	return -1;
+}
+
+/**
+ * @class TextCopyHelper
+ *
+ * @param {Object} config
+ * @param {string} config.text - Value of the url input element
+ * @param {string} config.message - Tooltip text
+ * @param {HTMLElement} config.parentEl - Parent element
+ * @param {function} config.onDestroy - Optional, callback that will be ran on {@link destroy}
+ * @param {function} config.onCopy - Optional, callback that will be ran when the 'copy' event is triggered
+ */
 function TextCopyHelper(config) {
```

Comment:
Seems to be just a case of named arguments, I generally prefer them if there are a few arguments, helps avoid API anti-patterns such as the '[boolean trap](http://ariya.ofilabs.com/2011/08/hall-of-api-shame-boolean-trap.html)'.

Although that aside, isn't adding event listeners to the object a more standard pattern rather than passing in callbacks as optional params?

```JS
var obj = new Constructor({ my: args });
obj.on('destroy', function() {
});
obj.on('copy', function() {
});
```